velox/type/DecimalUtil.h

```diff
@@ -147,6 +148,43 @@ class DecimalUtil {
     return static_cast<TOutput>(rescaledValue);
   }
 
+  template <typename TInput, typename TOutput>
+  inline static std::optional<TOutput> rescaleFloating(
+      const TInput inputValue,
+      const int toPrecision,
+      const int toScale) {
+    auto toValue =
+        inputValue * static_cast<TInput>(DecimalUtil::kPowersOfTen[toScale]);
```

Comment:
If I understand correctly, `toValue` can not be larger than `static_cast<TInput>(std::numeric_limits<TOutput>::max())`, because the returned value is of `TInput` type. Same for negative case. Is below check still needed?

```
    if (toValue > static_cast<TInput>(std::numeric_limits<TOutput>::max()) ||
        toValue < static_cast<TInput>(std::numeric_limits<TOutput>::min())) {
      isOverflow = true;
    }
```

Reply:
No, `toValue` could be larger than `static_cast<TInput>(std::numeric_limits<TOutput>::max())`, because TInput may be double, and TOutput may be int64_t. It is very usual that a double value is bigger than int64_t.
For example, if `inputValue` is 9223372036854775808, `toScale` is 2 and TOutput is int64_t, then `toValue` should be 922337203685477580800 in double type, it is bigger than `static_cast<double>(std::numeric_limits<int64_t>::max())`'value 9223372036854775807.

Reply:
Thanks for the details. If TOutput is int128_t, do we still need this check? Because double value cannot be larger the maximum of int128_t. Maybe this part can be moved to below else branch for bigint handling.